### PR TITLE
Add wildcard loading to ros2 param load

### PR DESCRIPTION
**Pull request by v-lopez**
Related to #590 and #596

### CODE REVIEW

File: ros2param/ros2param/api/__init__.py [hunk 1]
```diff
@@ -137,21 +137,27 @@ def load_parameter_dict(*, node, node_name, parameter_dict):
...
-        load_parameter_dict(node=node, node_name=node_name,
-                            parameter_dict=value['ros__parameters'])
+        for k in param_keys:
+            value = param_file[k]
+            if type(value) != dict or 'ros__parameters' not in value:
+                raise RuntimeError('Invalid structure of parameter file for node {}'
+                                   'expected same format as provided by ros2 param dump'
+                                   .format(k))
+            load_parameter_dict(node=node, node_name=node_name,
+                                parameter_dict=value['ros__parameters'])
```
Comment (ivanpauno) [inline ros2param/ros2param/api/__init__.py:160]: This works a bit better if there's a wildcard parameter value and one for the node:

```suggestion
        param_dict = {}
        for k in param_keys:
            value = param_file[k]
            if type(value) != dict or 'ros__parameters' not in value:
                raise RuntimeError('Invalid structure of parameter file for node {}'
                                   'expected same format as provided by ros2 param dump'
                                   .format(k))
            param_dict.update(value['ros__parameters'])
        load_parameter_dict(node=node, node_name=node_name, parameter_dict=param_dict])
```

File: ros2param/ros2param/verb/load.py [hunk 2]
```diff
@@ -34,8 +34,11 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
         parser.add_argument(
             '--include-hidden-nodes', action='store_true',
             help='Consider hidden nodes as well')
-        arg = parser.add_argument(
+        parser.add_argument(
             'parameter_file', help='Parameter file')
+        parser.add_argument(
+            '--use-wildcard', action='store_true',
+            help='Load parameters in the \'/**\' namespace into the node')
```
Comment (ivanpauno) [inline ros2param/ros2param/verb/load.py:41]: The default behavior should be that parameters from the wildcard section are loaded.
If you want, there can be a `--no-use-wildcard` option

File: ros2param/ros2param/api/__init__.py (hunk 1 above)
Comment (v-lopez) [reply ros2param/ros2param/api/__init__.py]: Certainly, good catch!

File: ros2param/ros2param/verb/load.py (hunk 2 above)
Comment (v-lopez) [reply ros2param/ros2param/verb/load.py]: Makes sense, please check the latest changes.

**Review by ivanpauno — APPROVED** [review]
LGTM, thanks for the PR @v-lopez !!

**Comment by ivanpauno** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13722/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8594/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11434/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13795/)

**Comment by ivanpauno** [comment]
@v-lopez there are a few linter failures

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13723/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8595/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11435/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13796/)

**Pull request merged**